File: domain/src/test/java/mixter/domain/core/subscription/FakeSubscriptionRepository.java
package mixter.domain.core.subscription;

import org.assertj.core.util.Sets;

import java.util.Set;

public class FakeSubscriptionRepository implements SubscriptionRepository {
    private Set<Subscription> subscriptions = Sets.newHashSet();

    public void add(Subscription subscription) {
        subscriptions.add(subscription);
    }

    @Override
    public Subscription getById(SubscriptionId subscriptionId) {
        return subscriptions.stream().filter(subscription -> subscription.getId().equals(subscriptionId)).findFirst().get();
    }
}

File: domain/src/test/scala/mixter/domain/subscription/FakeSubscriptionRepository.scala
package mixter.domain.subscription

import org.scalatest.{Suite, SuiteMixin}

private[subscription] trait SubscriptionRepositoryFixture extends SuiteMixin {
  this: Suite =>
  def withSubscriptionRepository(test: FakeSubscriptionRepository => Any): Any = {
    val sessionRepository = new FakeSubscriptionRepository()
    test(sessionRepository)
  }
}

private[subscription] class FakeSubscriptionRepository extends SubscriptionRepository {
  private var subscriptions = Set.empty[Subscription]

  def add(subscription: Subscription): Unit = subscriptions += subscription

  override def getById(subscriptionId: SubscriptionId): Option[Subscription] =
    subscriptions.find(_.initialEvent.subscriptionId==subscriptionId)
}